services/osparc-gateway-server/tests/unit/test_utils.py

```diff
@@ -313,3 +333,48 @@ async def test_create_or_update_docker_secrets(
         filters={"label": f"cluster_id={fake_cluster.id}"}
     )
     assert len(secrets) == 0
+
+
+async def test_get_cluster_information(
+    docker_swarm,
+    async_docker_client: aiodocker.Docker,
+):
+    cluster_information = await get_cluster_information(async_docker_client)
+    assert cluster_information
+
+    # in testing we do have 1 machine, that is... this very host
+    assert len(cluster_information) == 1
+    assert socket.gethostname() in cluster_information
+
+
+async def test_get_empty_node_hostname(
+    docker_swarm,
+    async_docker_client: aiodocker.Docker,
+    fake_cluster: Cluster,
+    running_service,
+    faker: Faker,
+):
+    hostname = await get_next_empty_node_hostname(async_docker_client, fake_cluster)
```

Comment:
THOUGHT: it would be interesting to have a test that can be configured (e.g. some yaml) to emulates a real situation (e.g. a node drops or a connection fails) and see how the system reacts to it. It does not apply to this repo but in general. The more piece we add, the more complexity we need to handle. Having a place where we can "inoculate" a problem. Look for "fault injection testing"

Reply:
I agree, but I don't know how to do that. it seems docker-machine has been deprecated.
see [https://docs.docker.com/machine/](https://docs.docker.com/machine/). any idea?

Reply:
Seems to be a big community around this e.g. a first hit in gg gave https://pythonrepo.com/tag/fault-injection

Perhaps we could have a look at it